### ISSUE TITLE
Create TRADEOFFS.md

### ISSUE THREAD

**Issue by hzoo** [babel/babel#4959]
https://twitter.com/jordwalke/status/801655139849248768 via @jordwalke - I like it!

Using Babel/ES2015+ is definetely a tradeoff

## Positives

+ Don't have to worry about browser support (similar to jQuery but for syntax)
+ Might be faster than native syntax (for a while)
+ Possibly more concise easier to read source code
  +  Only if people are taught it, and using it effectively
+ Keep up to date with the latest version of JavaScript, adopt it incrementally and grow the team
+ Community backing (10+ million downloads/month, many libraries, frameworks, guides use it), and used by many companies: http://babeljs.io/users
+ Tries to follow the specification as much as possible: so when you stop using Babel, or use babel-preset-env to not compile it there won't be many issues if any at all.
  + Trying to participate in TC39 and get committee members involved (and attending meetings)
+ Use/support a tool that is helping shape the future of JavaScript without having to get involved specifically in the committee itself - can help implement proposals and give feedback.
+ Tooling chain supports making your own plugins for development purposes
+ What most people think of when talking about a JS compiler/transpiler: many languages on top of JS use Babel to compile down to ES5 like [Coffeescript](http://coffeescript.org/#transpilation) and [Fable](http://fable.io/)

---

### Inherent
- Larger payload/code output (slower to download/parse/execute)
  - This can be optimized against engines but always going to be larger than native
  - `loose` mode is less spec compliant much faster/smaller
  - using babel-preset-env means shipping native code when possible
- Having to have a build step at all / Longer build times
  - Can argue there's already a need to bundle/minify, and other things like JSX, types, etc
- Have to deal with potential bugs in Babel vs. only the bugs in the browser
  - May not be as much of an issue in reality with sufficient community coverage
  - Integrating official ECMA test262 tests may help a lot here
  - Engines actually test against Babel when building out new features like object rest spread and class properties and file issues.
- Issues in debugging: having to use source-maps vs none (might not be an issue if you are already bundling/minifying anyway)
- If you use a Stage X proposal, it may get removed later or changed in a significant way so it should be used with caution
  - Shouldn't be used in production unless you know what you're doing
  - May confuse people reading the source code/documentation
  - Incentive for us/community to build codemods to automatically convert between breaking changes of the proposal as well as removing the proposal syntax if a proposal gets removed entirely.
  - Using the proposal will give you experience and allow one to provide valuable feedback to the proposal champions.
- If you are going to use new syntax/compile step you might want to just use an entirely different language like Elm, Clojurescript, etc instead of "BabelScript"

### Can be Fixed
- Small core team of maintainers and contributors for as long as the project has been around
  - Maybe this is supposed to be a positive since it's still alive and used
- Almost none of those top companies are sponsoring/helping/getting involved in the project in any significant way for it's future (even issues) - right now only Behance (Adobe) for 50% my time and AMP (Google) for 20% time for @jridgewell.
- Currently the situation around polyfills (babel-polyfill, core-js, transform-runtime, babel-runtime, babel-preset-env `useBuiltIns` option, etc) sucks
- People complain Babel itself takes up too much space (and that there are pictures of guy fieri)
  - Mostly because of babel-runtime in each package: which was removed in v7, and mostly fixed with npm/yarn deduping
  - It's a devDep
- People complain about too many packages
  - Yes I know. Talking forever about `babel` npm package that has no/minimal config that does the right defaults and works in most situations (maybe eject behavior)

**Comment by xtuc** [comment]
Good idea. We could include some design docs https://github.com/babel/babel/tree/master/doc/design. Monorepo has some tradeoffs.

**Comment by hzoo** [comment]
Probably don't need to track this, will just do it when I get to do it

**Comment by hzoo** [comment]
I think I wrote enough to start a PR somewhere

**Comment by xtuc** [comment]
@hzoo do we still want this?

**Comment by Ali-Haggag7** [comment]
Hi @hzoo, I see this issue has been open for a while. I'd love to help by creating the TRADEOFFS.md file using the points mentioned in the description. Is it okay if I start working on this?
